cloudsync/provider.py

```diff
@@ -408,19 +408,24 @@ def split(self, path):
             return self.sep, path[index + 1:]
         return path[:index], path[index+1:]
 
-    def normalize_path(self, path: str):
-        """Used internally for comparing paths in a case and sep insensitive manner, as appropriate."""
-        norm_path = path.rstrip(self.sep)
-        if self.sep in ["\\", "/"]:
-            parts = re.split(r'[\\/]+', norm_path)
-        else:
-            parts = re.split(r'[%s]+' % self.sep, norm_path)
+    def normalize_path(self, path: str, pretty_leaf: bool = False):
+        """Used internally for comparing paths in a case and sep insensitive manner, as appropriate.
+
+        Args:
+            path: the path to normalize
+            pretty_leaf: when true, preserve case of path's leaf node
+        """
+        if self.alt_sep:
+            path = path.replace(self.alt_sep, self.sep)
+        sep_regexp = r"[\\]+" if self.sep == "\\" else f"[{self.sep}]+"
```

Comment:
Maybe you mean to use `re.escape()` here?

Reply:
Cool, that worked, and looks much better. Thanks!

The reason I had that code in there is because pytest goes crazy when I do something like f'[{sep}]+' when sep is \. But outside of pytest I did not see an issue with that statement.